commit 88d6041d070028ef31c52845966216004ebba3bb
Author: Revanth Rajashekar <revanth.rajashekar@intel.com>
Date:   Tue Mar 3 12:17:00 2020 -0700

    block: sed-opal: Change the check condition for regular session validity
    
    This patch changes the check condition for the validity/authentication
    of the session.
    
    1. The Host Session Number(HSN) in the response should match the HSN for
       the session.
    2. The TPER Session Number(TSN) can never be less than 4096 for a regular
       session.
    
    Reference:
    Section 3.2.2.1   of https://trustedcomputinggroup.org/wp-content/uploads/TCG_Storage_Opal_SSC_Application_Note_1-00_1-00-Final.pdf
    Section 3.3.7.1.1 of https://trustedcomputinggroup.org/wp-content/uploads/TCG_Storage_Architecture_Core_Spec_v2.01_r1.00.pdf
    
    Co-developed-by: Andrzej Jakowski <andrzej.jakowski@linux.intel.com>
    Signed-off-by: Andrzej Jakowski <andrzej.jakowski@linux.intel.com>
    Signed-off-by: Revanth Rajashekar <revanth.rajashekar@intel.com>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/block/opal_proto.h b/block/opal_proto.h
index 325cbba2465f..b486b3ec7dc4 100644
--- a/block/opal_proto.h
+++ b/block/opal_proto.h
@@ -36,6 +36,7 @@ enum opal_response_token {
 
 #define DTAERROR_NO_METHOD_STATUS 0x89
 #define GENERIC_HOST_SESSION_NUM 0x41
+#define FIRST_TPER_SESSION_NUM	4096
 
 #define TPER_SYNC_SUPPORTED 0x01
 #define MBR_ENABLED_MASK 0x10